example/use_cases/employer_university_flow/employer_university_flow/wallet.go

```diff
@@ -0,0 +1,126 @@
+package employer_university_flow
+
+import (
+	gocrypto "crypto"
+	"errors"
+	"fmt"
+	"sync"
+
+	"github.com/TBD54566975/ssi-sdk/credential"
+	"github.com/TBD54566975/ssi-sdk/crypto"
+	"github.com/TBD54566975/ssi-sdk/did"
+)
+
+// A sample wallet
+// This would NOT be how it would be stored in production
+// But serves for demonstrative purposes
+// This holds the assigned dids
+// private keys
+// and vCs
+type SimpleWallet struct {
+	vCs  map[string]*credential.VerifiableCredential
+	keys map[string]gocrypto.PrivateKey
+	dids map[string]string
+	mux  *sync.Mutex
+}
+
+func NewSimpleWallet() *SimpleWallet {
+	return &SimpleWallet{
+		vCs:  make(map[string]*credential.VerifiableCredential),
+		mux:  &sync.Mutex{},
+		dids: make(map[string]string),
+		keys: make(map[string]gocrypto.PrivateKey),
+	}
+}
+
+// Adds a Private Key to a wallet
+func (s *SimpleWallet) AddPrivateKey(k string, key gocrypto.PrivateKey) error {
+	s.mux.Lock()
+	defer s.mux.Unlock()
+	if _, ok := s.keys[k]; !ok {
+		s.keys[k] = key
+	} else {
+		return errors.New("Already an entry")
+	}
+	return nil
+}
+
+func (s *SimpleWallet) AddDIDKey(k string, key string) error {
+	s.mux.Lock()
+	defer s.mux.Unlock()
+	if _, ok := s.dids[k]; !ok {
+		s.dids[k] = key
+	} else {
+		return errors.New("Already an entry")
+	}
+	return nil
+}
+
+func (s *SimpleWallet) GetDID(k string) (string, error) {
+	s.mux.Lock()
+	defer s.mux.Unlock()
+	if v, ok := s.dids[k]; ok {
+		return v, nil
+	} else {
+		return "", errors.New("Not found")
+	}
+	return "", nil
+}
+
+func (s *SimpleWallet) AddCredentials(cred *credential.VerifiableCredential) error {
+	s.mux.Lock()
+	defer s.mux.Unlock()
+	if _, ok := s.vCs[cred.ID]; !ok {
+		s.vCs[cred.ID] = cred
+	} else {
+		return errors.New("Duplicate Credential. Could not add.")
+	}
+	return nil
+}
+
+// In the simple wallet
+// Stores a DID for a particular user and
+// adds it to the registry
+func (s *SimpleWallet) Init(keyType string) error {
+
+	s.mux.Lock() // TODO: remove the muxes?
```

Comment:
what's the value here?

Reply:
in the muxing? preventing concurrent override of k:v to prevent duplicate creds. For example: if I concurrently added VC with 1, they would override eachother in the hashmap.